docs/concepts/storage/volumes.md

```diff
@@ -434,7 +434,10 @@ See the [iSCSI example](https://github.com/kubernetes/examples/tree/{{page.githu
 
 ### local
 
-This volume type is alpha in 1.7.
+{% assign for_k8s_version="v1.7" %}{% include feature-state-alpha.md %}
+
+This alpha feature requires the `PersistentLocalVolumes` feature gate to be
```

Comment:
PersistentLocalVolumes became a thing since v1.9, how come the 'local' volume type is available since v1.7?

Reply:
PersistentLocalVolumes was a new feature added in 1.7.

VolumeScheduling is a new feature in 1.9.

Reply:
They are considered two separate features because the VolumeScheduling feature can theoretically work for more than just local volumes.

Reply:
Hm this is even messier though.  Starting in 1.9, the VolumeScheduling feature gate is required if you also need the PersistentLocalVolumes feature.

Reply:
Ya. Better clearly state the requirement to clean the mess, NOW

Reply:
Sorry can you clarify?  I added a note stating that starting in 1.9, the VolumeScheduling feature gate is also required

Reply:
Just did a search over the source code. It looks to me that the `VolumeScheuling` feature gate has to be enabled on the following components:

- apiserver: for validation
- kcm: for pv controller
- scheduler: for scheduler behavior changes

For `PersistentLocalVolumes`, it has to be enabled on the following components:

- apiserver: for validation
- kubelet: for mounter
- scheduler: for node affinity checking?

Reply:
Yes that's correct.  But like I mentioned in the other PR, should we just recommend that that the feature gates be set on all components?  It would be a pain for an administrator to keep track of which feature gates are enabled where for each feature.